src/aiodynamo/http/aiohttp.py

```diff
@@ -1,18 +1,39 @@
 from dataclasses import dataclass
-from typing import Any, Dict, Mapping, Optional
+from functools import wraps
+from typing import Any, Dict, Optional
 
 from aiodynamo.types import Timeout
-from aiohttp import ClientSession
+from aiohttp import ClientError, ClientResponseError, ClientSession
 from yarl import URL
 
 from ..errors import exception_from_response
 from .base import HTTP, Headers, RequestFailed
 
 
+def wrap_errors(coro):
+    @wraps(coro)
+    async def wrapper(*args, **kwargs):
+        try:
+            return await coro(*args, **kwargs)
+        except ClientResponseError as exc:
+            raise RequestFailed(
+                exc.request_info.url,
+                exc.status,
+                exc.message,
+                exc.request_info.headers,
+                None,
+            )
+        except ClientError:
+            raise RequestFailed()
```

Comment:
I wonder if there's some info...
Would something like `... as e: raise R(str(e))` or `repr` be useful?
For example so that user can distinguish error cause, dns vs connection.

On the other hand, exceptions are chained by default, so perhaps there's enough information there already? 🤔

Reply:
> On the other hand, exceptions are chained by default, so perhaps there's enough information there already? 🤔

that was my thinking.